pkg/netstore/netstore_test.go

```diff
@@ -77,6 +93,12 @@ func TestNetstoreRetrieval(t *testing.T) {
 // whenever it is found locally.
 func TestNetstoreNoRetrieval(t *testing.T) {
 	retrieve, store, nstore := newRetrievingNetstore(nil, noopValidStamp)
+	defer func() {
+		err := nstore.Close()
```

Comment:
i would add this directly inside `newRetrievingNetstore` inside a `t.Cleanup()` (you'll need to pass the test helper). Then this change becomes redundant across all the individual tests